File: Golang/Hand of Straights.go
func isNStraightHand(hand []int, groupSize int) bool {
    l := len(hand)
	if l%groupSize != 0 {
		return false
	}
	sort.Ints(hand)

	for i := 0; i < l; i++ {
		if hand[i] == -1 {
			continue
		}
		k := 1
		tmp := hand[i]
		hand[i] = -1
		for j := i + 1; j < l && k < groupSize; j++ {
			if tmp+k == hand[j] && hand[j] != -1 {
				hand[j] = -1
				k++
			}
		}
		if k != groupSize {
			return false
		}
	}
	return true
}

File: Python3/Hand of Straights.py
class Solution:
    def isNStraightHand(self, hand: List[int], groupSize: int) -> bool:
        c = collections.Counter(hand)
        for i in sorted(c):
            if c[i] > 0:
                for j in range(groupSize)[::-1]:
                    c[i + j] -= c[i]
                    if c[i + j] < 0:
                        return False
        return True

#Count number of different cards to a map c
#Loop from the smallest card number.
#Everytime we meet a new card i, we cut off i - i + W - 1 from the counter.